.github/workflows/cache.yml

```diff
@@ -62,37 +62,18 @@ jobs:
           persist-credentials: false
 
       - name: Setup ASDF
-        uses: asdf-vm/actions/setup@v3
-        with:
-          asdf_branch: ${{ vars.ASDF_BRANCH }}
-
-      - name: Cache ASDF
-        uses: actions/cache/restore@v4
-        id: asdf-cache
-        with:
-          # https://github.com/asdf-vm/asdf/blob/master/.gitignore
-          path: |
-            ~/.asdf/installs
-            ~/.asdf/plugins
-            ~/.asdf/shims
-          key: ${{ runner.os }}-asdf-${{ hashFiles('.tool-versions') }}
-
-      - name: Install ASDF Tools
-        uses: asdf-vm/actions/install@v3
-        # See https://github.com/asdf-vm/actions/issues/445
-        if: ${{ steps.asdf-cache.outputs.cache-hit != 'true' }}
-        with:
-          asdf_branch: ${{ vars.ASDF_BRANCH }}
-
-      - name: Reshim ASDF
-        shell: bash
-        run: asdf reshim
+        uses: ./.github/actions/setup-asdf
 
       - name: Run dialyzer for caching
         run: bin/bix ex-dialyzer
 
       - name: Run tests for caching
-        run: bin/bix ex-test-deep
+        shell: bash
+        env:
+          MIX_ENV: test
+        run: |
+          pushd platform_umbrella
+          mix compile --force
```

Comment:
```suggestion
          pushd platform_umbrella
          mix "do", deps.get, compile --force
```

Reply:
If the deps change then the cache won't have the deps there so we always need to ask to get the deps before compiling.

Reply:
Yeah, I was thinking the dialyzer step beforehand would do it but it wouldn't pull test-only dependencies. :facepalm: